public/app/features/variables/query/operators.ts

```diff
@@ -18,87 +18,86 @@ import { getTemplatedRegex, toKeyedVariableIdentifier, toVariablePayload } from
-              textIndex = index;
-            }
-
-            if (fieldName === 'value' && field.type === FieldType.string && valueIndex === -1) {
-              valueIndex = index;
-            }
-
-            if (
-              fieldName === 'expandable' &&
-              (field.type === FieldType.boolean || field.type === FieldType.number) &&
-              expandableIndex === -1
-            ) {
-              expandableIndex = index;
-            }
-          }
-        }
+  if (areMetricFindValues(frames)) {
+    return frames;
+  }
 
-        if (stringIndex === -1) {
-          throw new Error("Couldn't find any field of type string in the results.");
-        }
+  const processedDataFrames = getProcessedDataFrames(frames);
+  const metrics: MetricFindValue[] = [];
+
+  let valueIndex = -1;
+  let textIndex = -1;
+  let stringIndex = -1;
+  let expandableIndex = -1;
+
+  for (const frame of processedDataFrames) {
+    for (let index = 0; index < frame.fields.length; index++) {
+      const field = frame.fields[index];
+      const fieldName = getFieldDisplayName(field, frame, frames).toLowerCase();
+
+      if (field.type === FieldType.string && stringIndex === -1) {
+        stringIndex = index;
+      }
+
+      if (fieldName === 'text' && field.type === FieldType.string && textIndex === -1) {
+        textIndex = index;
+      }
+
+      if (fieldName === 'value' && field.type === FieldType.string && valueIndex === -1) {
+        valueIndex = index;
+      }
+
+      if (
+        fieldName === 'expandable' &&
+        (field.type === FieldType.boolean || field.type === FieldType.number) &&
+        expandableIndex === -1
+      ) {
+        expandableIndex = index;
+      }
+    }
+  }
 
-        for (const frame of frames) {
-          for (let index = 0; index < frame.length; index++) {
-            const expandable = expandableIndex !== -1 ? frame.fields[expandableIndex].values[index] : undefined;
-            const string = frame.fields[stringIndex].values[index];
-            const text = textIndex !== -1 ? frame.fields[textIndex].values[index] : null;
-            const value = valueIndex !== -1 ? frame.fields[valueIndex].values[index] : null;
-
-            if (valueIndex === -1 && textIndex === -1) {
-              metrics.push({ text: string, value: string, expandable });
-              continue;
-            }
-
-            if (valueIndex === -1 && textIndex !== -1) {
-              metrics.push({ text, value: text, expandable });
-              continue;
-            }
-
-            if (valueIndex !== -1 && textIndex === -1) {
-              metrics.push({ text: value, value, expandable });
-              continue;
-            }
-
-            metrics.push({ text, value, expandable });
-          }
-        }
+  if (stringIndex === -1) {
+    throw new Error("Couldn't find any field of type string in the results.");
+  }
 
-        return metrics;
-      })
-    );
+  for (const frame of processedDataFrames) {
```

Comment:
This is the main and only change (sorry for the diff), before we used `frames` here instead of `processedDataFrames` but `frames` could be anything and we purposefully processed them into oridnary data frames so it did not make sense to not use the processedDataFrames here.